### PR TITLE
Make it possible to run node e2e with GCI.

### PR DESCRIPTION
**Pull request by vishh**

### CODE REVIEW

**Comment by vishh** [comment]
cc @yinghan @matchstick 

Excepting summary API tests, all other tests are passing. I hope to enable GCI as part of node e2e suite soon!

Thanks @Random-Liu for all the input!

File: test/e2e_node/e2e_service.go [hunk 1]
```diff
@@ -179,7 +180,16 @@ func (es *e2eService) startEtcd() (*killCmd, error) {
 		return nil, err
 	}
 	es.etcdDataDir = dataDir
-	cmd := exec.Command("etcd")
+	etcdPath, err := exec.LookPath("etcd")
```
Comment (Random-Liu) [inline test/e2e_node/e2e_service.go]: A few comments are appreciated. :)

File: test/e2e_node/e2e_node_suite_test.go [hunk 2]
```diff
@@ -135,7 +135,8 @@ var _ = AfterSuite(func() {
 func maskLocksmithdOnCoreos() {
 	data, err := ioutil.ReadFile("/etc/os-release")
 	if err != nil {
-		glog.Fatalf("Could not read /etc/os-release: %v", err)
+		glog.Infof("Could not read /etc/os-release: %v", err)
```
Comment (Random-Liu) [inline test/e2e_node/e2e_node_suite_test.go]: A few comments here to explain what happened? :)

**Comment by Random-Liu** [comment]
LGTM with nits.

File: test/e2e_node/e2e_service.go (hunk 1 above)
Comment (vishh) [inline test/e2e_node/e2e_service.go]: Added comments and docs.

File: test/e2e_node/e2e_node_suite_test.go (hunk 2 above)
Comment (vishh) [inline test/e2e_node/e2e_node_suite_test.go]: Done

**Comment by vishh** [comment]
@Random-Liu comments addressed PTAL.

**Comment by Random-Liu** [comment]
LGTM

**Comment by vishh** [comment]
Bumping up priority since this blocks other PRs and is critical for validating GCI with k8s.

**Pull request merged**